connection.go

```diff
@@ -10,14 +10,38 @@ type Client interface {
 	Call(result interface{}, method string, args ...interface{}) error
 
 	Subscribe(ctx context.Context, namespace string, channel interface{}, args ...interface{}) (*rpc.ClientSubscription, error)
+
+	MetaData(cache bool) (*MetadataVersioned, error)
+}
+
+type client struct {
+	*rpc.Client
+
+	// m is the metadata cache to prevent unnecessary requests
+	m *MetadataVersioned
+}
+
+func (c *client) MetaData(cache bool) (m *MetadataVersioned, err error) {
```

Comment:
According to my understanding, the only thing that could invalidate metadata is a chain upgrade. Since existing functions will work as expected as long as the upgrade is backward compatible, it's not necessary to invalidate the cache in that case. But ideal would be to invalidate cache upon detecting that 
1. metadata was not matching the logic or
2. some RPC call failed because of incorrect construction.
Best is to implement that logic once we have more RPCs implemented, I think. For now this cache helps in avoiding unnecessary metadata calls into the runtime most of the time.